dist/profile/manifests/kubernetes/resources/datadog.pp

```diff
@@ -0,0 +1,62 @@
+# Deploy datadog resources on kubernetes cluster
+class profile::kubernetes::resources::datadog (
+    $apiKey = base64('encode', $::datadog_agent::api_key, 'strict')
+  ){
+  include ::stdlib
+  include profile::kubernetes::params
+  include profile::kubernetes::kubectl
+
+  file { 'datadog_dir':
+    ensure => 'directory',
+    owner  => $profile::kubernetes::params::user,
+    path   => "${profile::kubernetes::params::resources}/datadog",
+  }
+
+  file { 'datadog_secret':
+    ensure  => 'present',
+    content => template("${module_name}/kubernetes/resources/datadog/secret.yaml.erb"),
+    path    => "${profile::kubernetes::params::resources}/datadog/secret.yaml",
+    owner   => $profile::kubernetes::params::user,
+  }
+
+  file { 'datadog_daemonset':
```

Comment:
The `name` ("datadog_daemonset") shoudl be changed to `"${profile::kubernetes::params::resources}/datadog/daemonset.yaml"` instead of using `path` below. The `path` parameter is unnecessary.

Same applies to the other `file` resources in this pull request